src/main/java/mops/gruppen2/domain/service/EventStoreService.java

```diff
@@ -176,35 +174,80 @@ private static String getEventType(Event event) {
      * @return GruppenIds (UUID) als Liste
      */
     List<UUID> findExistingGroupIds() {
-        List<Event> createEvents = findLatestEventsFromGroupsByType("CreateGroupEvent",
-                                                                    "DeleteGroupEvent");
+        List<Event> createEvents = findLatestEventsFromGroupsByType(CREATEGROUP,
+                                                                    DESTROYGROUP);
 
         return createEvents.stream()
                            .filter(event -> event instanceof CreateGroupEvent)
-                           .map(Event::getGroupId)
+                           .map(Event::getGroupid)
                            .collect(Collectors.toList());
     }
 
+    public List<UUID> findPublicGroupIds() {
+        List<UUID> groups = findExistingGroupIds();
+        List<Event> typeEvents = findLatestEventsFromGroupsByType(SETTYPE);
+
+        typeEvents.removeIf(event -> ((SetTypeEvent) event).getType() == Type.PRIVATE);
+        typeEvents.removeIf(event -> !groups.contains(event.getGroupid()));
+
+        return typeEvents.stream()
+                         .map(Event::getGroupid)
+                         .collect(Collectors.toList());
+    }
+
+    public List<UUID> findLectureGroupIds() {
+        List<UUID> groups = findExistingGroupIds();
+        List<Event> typeEvents = findLatestEventsFromGroupsByType(SETTYPE);
+
+        typeEvents.removeIf(event -> ((SetTypeEvent) event).getType() != Type.LECTURE);
+        typeEvents.removeIf(event -> !groups.contains(event.getGroupid()));
+
+        return typeEvents.stream()
+                         .map(Event::getGroupid)
+                         .collect(Collectors.toList());
+    }
+
     /**
      * Liefert Gruppen-Ids von existierenden (ungelöschten) Gruppen, in welchen der User teilnimmt.
      *
+     * <p>
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [<p> tag should be placed immediately before the first word, with no space after.](https://app.codacy.com/manual/tobi_2/gruppenfindung/pullRequest?prid=5356872)